File: stack.py
class Stack:
    class Node:
        def __init__(self, value):
            self.value = value
            self.next = None

    def __init__(self):
        self.top = None

    def __iter__(self):
        self.start = self.top
        return self

    def __next__(self):
        if self.start.next:
            self.start = self.start.next
            return self.start
        else:
            raise StopIteration

    def push(self, value):
        if not self.top:
            self.top = self.Node(value)
        else:
            new_data = self.Node(value)
            new_data.next = self.top
            self.top = new_data

    def pop(self):
        deleted_data = self.top
        self.top = self.top.next
        return deleted_data.value

    def peek(self):
        return self.top

    def __repr__(self):
        var = "["
        for i in self:
            var += " " + str(i.value)
        var += " ]"
        return var

File: data structure/stack.c
#include<stdlib.h>
#define Node struct node

Node{
    float data;
    Node *next;
};

Node *top = NULL;

void push(float value){
    Node *new_data = (Node*) malloc(sizeof(Node));
    new_data -> data = value;
    if(!top){
        new_data -> next = NULL;
        top = new_data;
    }
    else{
        new_data -> next = top;
        top = new_data;
    }
}

float pop(){
    float data = top -> data;
    top = top -> next;
    return data;
}

float peek(){
    return top -> data;
}
